src/engine/enginebuffer.h

```diff
@@ -380,6 +385,7 @@ class EngineBuffer : public EngineObject {
 
     SlipModeState m_slipModeState;
 
+    // Track samples are always given assuming a stereo track
     ControlObject* m_pTrackSamples;
     ControlObject* m_pTrackSampleRate;
```

Comment:
I think we should either return the correct sample rate (considering the stem channel count), or deprecate the COs `track_samples` and `track_samplerate` and add new COs `track_frames` and `track_framerate`.

Reply:
I'd be in favour of deprecating it and introducing frame count only. There is many places which relies on this CO and device by two to compute frame count, so I don't think we can return the correct sample rate without having to introduce a second CO proxy which will have a performance cost. What would be the value to return the "correct" sample rate tho?

Reply:
I think we should discuss this on Zulip - to get a common understanding of the terms.

Reply:
Sure. Also just FYI on the current behaviour just so that isn't a blocker for this PR, mono tracks would get a "fake" sample count as if they were stereo track, so this comment is only documenting a current behaviour

Reply:
Created the issue #13168 to address in the future